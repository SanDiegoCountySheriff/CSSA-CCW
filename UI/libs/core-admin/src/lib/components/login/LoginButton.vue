<!-- eslint-disable @intlify/vue-i18n/no-raw-text -->
<template>
  <div>
    <template v-if="authStore.getAuthState.isAuthenticated">
      <v-list-item link>
        <v-list-item-icon>
          <v-icon> mdi-account-circle </v-icon>
        </v-list-item-icon>
        <v-list-item-content>
          <v-list-item-title>
            {{ authStore.getAuthState.userName }}
          </v-list-item-title>
          <v-list-item-subtitle>
            {{ authStore.getAuthState.userEmail }}
          </v-list-item-subtitle>
        </v-list-item-content>
      </v-list-item>
    </template>
    <div v-else>
      <v-list-item
        link
        @click="handleLogIn"
      >
        <v-list-item-icon>
          <v-icon> mdi-account-circle </v-icon>
        </v-list-item-icon>
        <v-list-item-title> Login </v-list-item-title>
      </v-list-item>
    </div>
  </div>
</template>

<script setup lang="ts">
import { onMounted } from 'vue';
import auth from '@shared-ui/api/auth/authentication';
import { useAuthStore } from '@shared-ui/stores/auth';
import { useQuery } from '@tanstack/vue-query';

const authStore = useAuthStore();

onMounted(() => {
  if (authStore.getAuthState.isAuthenticated) {
    useQuery(['verifyEmail'], authStore.postVerifyUserApi);
  }
});

// eslint-disable-next-line @typescript-eslint/no-unused-vars
async function logout() {
  await auth.signOut();
}

function handleLogIn() {
  auth.signIn();
}
</script>

<style lang="scss" scoped>
.v-list-item {
  &__title {
    text-align: left;
  }

  &__subtitle {
    text-align: left;
  }
}
</style>
