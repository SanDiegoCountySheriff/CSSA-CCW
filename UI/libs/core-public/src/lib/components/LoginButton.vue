<template>
  <div>
    <Button
      color="primary"
      text="Login"
      @click="handleLogIn"
    >
    </Button>
  </div>
</template>

<script setup lang="ts">
import auth from '@core-public/api/auth/authentication';
import Button from '@shared-ui/components/Button.vue';

function handleLogIn() {
  auth.signIn();
}
</script>
