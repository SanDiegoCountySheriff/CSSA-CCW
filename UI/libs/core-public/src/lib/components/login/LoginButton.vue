<template>
  <div>
    <template v-if="authStore.getAuthState.isAuthenticated">
      <div class="d-flex">
        <h3 class="text-center white--text font-weight-bold mr-4 ml-1">
          {{ authStore.getAuthState.userName }}
        </h3>
        <v-btn
          v-if="authStore.getAuthState.isAuthenticated"
          aria-label="Sign out of application"
          @click="signOut"
          class="mr-4 ml-1"
          color="primary lighten-2"
          small
        >
          <!--eslint-disable-next-line vue/singleline-html-element-content-newline -->
          <v-icon
            v-if="$vuetify.breakpoint.mdAndDown"
            class="pr-1 white--text"
          >
            mdi-logout-variant
          </v-icon>
          <span
            v-if="$vuetify.breakpoint.mdAndUp"
            class="white--text"
            >{{ $t('Sign out') }}</span
          >
        </v-btn>
      </div>
    </template>
    <div v-else>
      <Button
        color="primary"
        text="Login"
        @click="handleLogIn"
      >
      </Button>
    </div>
  </div>
</template>

<script setup lang="ts">
import Button from '@shared-ui/components/Button.vue';
import auth from '@shared-ui/api/auth/authentication';
import { onMounted } from 'vue';
import { useAuthStore } from '@shared-ui/stores/auth';
import { useBrandStore } from '@shared-ui/stores/brandStore';
import useInterval from '@shared-ui/composables/useInterval';
import { useQuery } from '@tanstack/vue-query';

const authStore = useAuthStore();
const brandStore = useBrandStore();

onMounted(() => {
  if (authStore.getAuthState.isAuthenticated) {
    useQuery(['verifyEmail'], authStore.postVerifyUserApi);
    useInterval(auth.tokenInterval, brandStore.getBrand.refreshTokenTime);
  }
});

async function signOut() {
  await auth.signOut();
}

function handleLogIn() {
  auth.signIn();
}
</script>
