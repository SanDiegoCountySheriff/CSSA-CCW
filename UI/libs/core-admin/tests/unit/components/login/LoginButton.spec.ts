/* eslint-disable @typescript-eslint/ban-ts-comment */
import LoginButton from '@core-admin/components/login/LoginButton.vue';
import Vuetify from 'vuetify';
import { createLocalVue, mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

const localVue = createLocalVue();
const pinia = createTestingPinia();
const tMock = {
  $t: text => text,
};

describe('LoginButton', () => {
  let vuetify;
  let wrapper;
  beforeEach(() => {
    vuetify = new Vuetify();
    //@ts-ignore
    wrapper = mount(LoginButton, {
      localVue,
      vuetify,
      pinia,
      mocks: tMock,
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });

  it('should match the snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render the Login button', () => {
    expect(wrapper.find('.v-list-item__title').text()).toBe('Login');
  });
});
