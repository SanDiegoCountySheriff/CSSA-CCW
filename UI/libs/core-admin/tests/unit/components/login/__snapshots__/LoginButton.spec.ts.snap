// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should match the snapshot 1`] = `
<div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary" color="primary"><span class="v-btn__content">
  Login
</span></button></div>
`;
